attestation-container/attestation-container.go

```diff
@@ -68,6 +76,21 @@ func main() {
 	flag.Parse()
 	validateFlags()
 
+	if *endorsementServer == "" {
+		log.Printf("Reading report endorsement from environment variable %s", *endorsementEnvironmentVariable)
+		endorsementEnvironmentValue = new(attest.ACIEndorsements)
+		var err error
+		endorsementEnvironment, ok := os.LookupEnv(*endorsementEnvironmentVariable)
+		if !ok {
+			log.Fatalf("Endorsement environment variable %s is not specified (or set --endorsement-server)", *endorsementEnvironmentVariable)
+		}
+
+		*endorsementEnvironmentValue, err = attest.ParseEndorsementACI(endorsementEnvironment)
+		if err != nil {
+			log.Fatal(err)
+		}
```

Comment:
I want to make `main` as small as possible and `attest` package as reusable. (I think `main` package basically should focus only on gRPC stuff.) In that sense how about making these lines a function in `attest` package like `attest.FetchAttestationEndorsementEnvVar(*endorsementEnvironmentVariable)`?

Reply:
That's what I originally did but then split it out to be able to unit test. I'll move the fetching of the envvar to the `attest` package.

Reply:
Is it possible to write unit test by using `os.Setenv(envvarName, contentsReadFromFile)` at the begining of the test function?
It may require to separate each test case to different  test function like `TestInvalidJson()` etc though

Reply:
something like:
```go
func TestInvalidJson() {
  contentsReadFromFile := readFile()
  contentsReadFromFile = breakJsonFormat(contentsReadFromFile)
  os.Setenv(ENV_VAR_NAME, contentsReadFromFile)
  // do check
}
```

Reply:
Change in [072fe0b](https://github.com/microsoft/CCF/pull/4948/commits/072fe0b8759b878b50b6b470d55b92f0155acbe7) looks good to me.